src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1900,85 +1900,102 @@ impl Cage {
-                (F_SETFD, arg) if arg >= 0 => {
-                    if arg & O_CLOEXEC != 0 {
-                        *flags |= O_CLOEXEC;
-                    } else {
-                        *flags &= !O_CLOEXEC;
+                //matching the tuple
+                match (cmd, arg) {
+                    //because the arg parameter is not used in certain commands, it can be anything (..)
+                    //currently, O_CLOEXEC is the only defined file descriptor flag, thus only this flag is
+                    //masked when using F_GETFD or F_SETFD
+                    (F_GETFD, ..) => *flags & O_CLOEXEC,
+                    // set the flags but make sure that the flags are valid
+                    (F_SETFD, arg) if arg >= 0 => {
+                        if arg & O_CLOEXEC != 0 {
+                            *flags |= O_CLOEXEC;
+                        } else {
+                            *flags &= !O_CLOEXEC;
+                        }
+                        0
                     }
-                    0
-                }
-                (F_GETFL, ..) => {
-                    //for get, we just need to return the flags
-                    *flags & !O_CLOEXEC
-                }
-                (F_SETFL, arg) if arg >= 0 => {
-                    *flags |= arg;
-                    0
-                }
-                (F_DUPFD, arg) if arg >= 0 => self._dup2_helper(&filedesc_enum, arg, false),
-                //TO DO: implement. this one is saying get the signals
-                (F_GETOWN, ..) => {
-                    0 //TO DO: traditional SIGIO behavior
-                }
-                (F_SETOWN, arg) if arg >= 0 => {
-                    0 //this would return the PID if positive and the process group if negative,
-                      //either way do nothing and return success
+                    //F_GETFL should return file access mode and the file status flags, thus excluding
+                    //the file creation flags. It is not clear why O_CLOEXEC is the only masked-out flag,
+                    //when there are other file creation flags, like O_CREAT, O_DIRECTORY, etc. 
+                    (F_GETFL, ..) => {
+                        //for get, we just need to return the flags
+                        *flags & !O_CLOEXEC
+                    }
+                    (F_SETFL, arg) if arg >= 0 => {
+                        *flags |= arg;
+                        0
+                    }
+                    (F_DUPFD, arg) if arg >= 0 => self._dup2_helper(&filedesc_enum, arg, false),
+                    //TO DO: implement. this one is saying get the signals
+                    (F_GETOWN, ..) => {
+                        0 //TO DO: traditional SIGIO behavior
+                    }
+                    (F_SETOWN, arg) if arg >= 0 => {
+                        0 //this would return the PID if positive and the process group if negative,
```

Comment:
Would it set this or return it?  Why do we do nothing?

Reply:
When I was talking to Yash, he suggested me that I should not focus on things that are not yet implemented, and that is why I left this part with F_SETOWN and F_GETOWN commands untouched. Should I work on them?